core/src/test/java/org/elasticsearch/indices/memory/breaker/CircuitBreakerServiceIT.java

```diff
@@ -337,4 +351,69 @@ public void testCustomCircuitBreakerRegistration() throws Exception {
         }
         assertThat(breaks, greaterThanOrEqualTo(1));
     }
+
+    public void testLimitsRequestSize() throws Exception {
+        ByteSizeValue inFlightRequestsLimit = new ByteSizeValue(8, ByteSizeUnit.KB);
+        if (noopBreakerUsed()) {
+            logger.info("--> noop breakers used, skipping test");
+            return;
+        }
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+
+        Settings limitSettings = settingsBuilder()
+            .put(HierarchyCircuitBreakerService.IN_FLIGHT_REQUESTS_CIRCUIT_BREAKER_LIMIT_SETTING.getKey(), inFlightRequestsLimit)
+            .build();
+
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setTransientSettings(limitSettings));
```

Comment:
do we want to do this just before indexing? to make sure we don't block anything else?